doc/contribute/guidelines.rst

```diff
@@ -643,53 +649,141 @@ workflow here:
    Additional information about the CI system can be found in
    `Continuous Integration`_.
 
+.. _commit-guidelines:
+
+Commit Message Guidelines
+*************************
+
+Changes are submitted as Git commits. Each commit has a *commit
+message* describing the change. Acceptable commit messages look like
+this:
+
+.. code-block:: none
 
-Commit Guidelines
-*****************
+   [area]: [summary of change]
 
-Changes are submitted as Git commits. Each commit message must contain:
+   [Commit message body (must be non-empty)]
 
-* A short and descriptive subject line that is less than 72 characters,
-  followed by a blank line. The subject line must include a prefix that
-  identifies the subsystem being changed, followed by a colon, and a short
-  title, for example:  ``doc: update wiki references to new site``.
-  (If you're updating an existing file, you can use
-  ``git log <filename>`` to see what developers used as the prefix for
-  previous patches of this file.)
+   Signed-off-by: [Your Full Name] <[your.email@address]>
 
-* A change description with your logic or reasoning for the changes, followed
-  by a blank line. (Every single line has to be less than 75 characters.)
+You need to change text in square brackets (``[like this]``) above to
+fit your commit.
 
-* A Signed-off-by line, ``Signed-off-by: <name> <email>`` typically added
-  automatically by using ``git commit -s``
+Examples and more details follow.
+
+Example
+=======
+
+Here is an example of a good commit message.
+
+.. code-block:: none
 
-* If the change addresses an issue, include a line of the form::
+   drivers: sensor: abcd1234: fix bus I/O error handling
 
-      Fixes #<issue number>.
+   The abcd1234 sensor driver is failing to check the flags field in
+   the response packet from the device which signals that an error
+   occurred. This can lead to reading invalid data from the response
+   buffer. Fix it by checking the flag and adding an error path.
 
+   Signed-off-by: Zephyr Developer <z.developer@example.com>
 
-All changes and topics sent to GitHub must be well-formed, as described above.
+[area]: [summary of change]
+===========================
+
+This line is called the commit's *title*. Titles must be:
+
+* one line
+* less than 72 characters long
+* followed by a completely blank line
+
+[area]
```

Comment:
area might not always fit, for example someone fixing a typo or a stlye issue across the tree will not be able to provide a specific area. Area below is defined as a subsystem. I have seen people use `treewide` in many places, but I do not think this one is right either, maybe we should add:

    In cases where a change touches many subsystems and areas, for example when fixing style issues, typos or 
    compliance issues, you should provide the context or the general theme of the change, such as `cleanup:`, 
    `style:`, `typos:`, etc.

Something long those lines.

Reply:
Changing to:

```
  The ``[area]`` prefix usually identifies the area of code
  being changed. It can also identify the change's wider
  context if multiple areas are affected.

  Here are some examples:

  * ``doc: ...`` for documentation changes
  * ``drivers: foo:`` for ``foo`` driver changes
  * ``Bluetooth: Shell:`` for changes to the Bluetooth shell
  * ``net: ethernet:`` for Ethernet-related networking changes
  * ``treewide:`` for :ref:`treewide-changes`
  * ``style:`` for code style changes
```